CsvParser/csvParser.go

```diff
@@ -94,47 +98,158 @@ func CsvParser(filePath string, numIncidents ...int) []incident.DefaultIncident
 	return defaultIncidents
 }
 
-//* This function takes in an array of 100 default incidents because that is what is provided by the csvParser above *//
-func convertDefaultIncidentToIncident(defaultIncidents []incident.DefaultIncident) []incident.Incident {
-	incidentArr := make([]incident.Incident, len(defaultIncidents), len(defaultIncidents))
-	for i := range defaultIncidents {
-		incidentArr[i] = &defaultIncidents[i]
-	}
-	return incidentArr
+//convert a default incident that implements our interface to an incident object designed to exist in a json file
+//It returns incident of type IncidentData to 
+func convertDefaultIncidentToIncidentData(i incident.DefaultIncident) incident.IncidentData {
+	gpStart, gpEnd := i.GetGoodPeriod()
+	bpStart, bpEnd := i.GetBadPeriod()
+	gMetric := i.GetGoodMetric()
+	bMetric := i.GetBadMetric()
+	severity := i.GetSeverity()
+	testsAffected := i.GetTestsAffected()
+	gpInfo := i.GetGoodPeriodInfo()
+	bpInfo := i.GetBadPeriodInfo()
+	iInfo := i.GetIncidentInfo()
+	locationString := i.GetLocation()
+	asn := i.GetASN()
+	inc := incident.IncidentData{gpStart, gpEnd, bpStart, bpEnd, gMetric, bMetric, asn, locationString, severity, testsAffected, gpInfo, bpInfo, iInfo}
+	return inc
 }
 
-func makeJsonObjFile(arr []incident.Incident) *os.File {
-	// numObjects determines how many incidents are stored in the json
-	const numObjects = 1
-	f, err := os.Create("incidents.json")
-	var objs [numObjects]incident.IncidentData
-	if err != nil {
-		log.Fatal(err)
-		return f
+//Takes in a location code and breaks it down into different location levels
+//Returns an array of string, where each string represents a location
+func parseLocationCode(locationCode string) []string {
+	locationCodesArr := make([]string, 0)
+
+	if (len(locationCode)) > 6 {
+		for i := 0; i < 6; i = i + 2 {
+			locationCodesArr = append(locationCodesArr, locationCode[i:i+2])
+		}
+
+		locationCodesArr = append(locationCodesArr, locationCode[6:])
+
+		return locationCodesArr
 	}
 
-	for i := 0; i < numObjects; i++ {
-		gpStart, gpEnd := arr[i].GetGoodPeriod()
-		bpStart, bpEnd := arr[i].GetBadPeriod()
-		gMetric := arr[i].GetGoodMetric()
-		bMetric := arr[i].GetBadMetric()
-		severity := arr[i].GetSeverity()
-		testsAffected := arr[i].GetTestsAffected()
-		gpInfo := arr[i].GetGoodPeriodInfo()
-		bpInfo := arr[i].GetBadPeriodInfo()
-		iInfo := arr[i].GetIncidentInfo()
-		inc := incident.IncidentData{gpStart, gpEnd, bpStart, bpEnd, gMetric, bMetric, severity, testsAffected, gpInfo, bpInfo, iInfo}
-		objs[i] = inc
+	for i := 0; i < len(locationCode); i = i + 2 {
+		locationCodesArr = append(locationCodesArr, locationCode[i:i+2])
 	}
-	bytes, err := json.Marshal(objs)
-	n, err := f.Write(bytes)
 
-	if err != nil {
-		log.Fatal(n)
-		log.Fatal(err)
-		f.Close()
-		return f
+	return locationCodesArr
+}
+
+//checks if a specific directory, which corresponds to a specifi location, has been made
```

Comment:
Change comment to "Check if a directory (corresponding to a specific location) has been made"